### PR TITLE
Doc Fixit: PHP update readme

### LINKED ISSUE #5631: Doc Fixit: update PHP README

### PR DESCRIPTION
**Pull request by stanley-cheung**
Fixes #5631 

Might be easier to directly view the document here:
- https://github.com/stanley-cheung/grpc/tree/php-update-readme/src/php
- https://github.com/stanley-cheung/grpc/tree/php-update-readme/examples/php

### CODE REVIEW

File: src/php/README.md [hunk 1]
```diff
@@ -39,15 +46,22 @@ Install the gRPC PHP extension
 sudo pecl install grpc-beta
 ```
 
+This will compile and install the gRPC PHP extension into the standard PHP extension directory. You should be able to run the [unit tests](#unit-tests), with the PHP extension installed.
+
+To run tests with generated stub code from `.proto` files, you will need the `composer`, `protoc` and `protoc-gen-php` binaries additionally. See sections [below](#generated-code-tests).
+
```
Comment (LisaFC) [inline src/php/README.md]: ...you will also need the `composer`, `protoc` and `protoc-gen-php` binaries. You can find out how to get these [below](#generated-code-tests).

File: src/php/README.md [hunk 2]
```diff
@@ -79,58 +88,96 @@ $ make
...
 ```sh
 $ cd grpc/src/php
+$ ./bin/run_tests.sh
+```
+
+## Generated Code Tests
+
+### Composer
+
+You need to install `composer`, to pull in some runtime dependencies based on the `composer.json` file.
```
Comment (LisaFC) [inline src/php/README.md]: You need to install `composer` to pull in some runtime dependencies based on the `composer.json` file.

File: src/php/README.md [hunk 3]
```diff
@@ -79,58 +88,96 @@ $ make
...
+
+Run unit tests
 
 ```sh
 $ cd grpc/src/php
+$ ./bin/run_tests.sh
+```
+
+## Generated Code Tests
+
```
Comment (LisaFC) [inline src/php/README.md]: Add:

This section specifies the prerequisites for running the generated code tests, as well as how to run the tests themselves.

File: src/php/README.md [hunk 4]
```diff
@@ -79,58 +88,96 @@ $ make
...
+$ cd grpc/src/php
 $ composer install
 ```
 
-## Unit Tests
+### Protobuf compiler
 
-Run unit tests
+You need the install the protobuf compiler, `protoc`, 3.0.0+.
```
Comment (LisaFC) [inline src/php/README.md]: Again if you don't have it already, you need to install the protobuf compiler `protoc`, version 3.0.0+.

File: src/php/README.md [hunk 5]
```diff
@@ -79,58 +88,96 @@ $ make
...
+
+If you had compiled the gRPC C core library from source above, the `protoc` binary should have been installed as well. In the case it wasn't, you can run the following commands to install it.
```
Comment (LisaFC) [inline src/php/README.md]: If you compiled the gRPC C core library from source above, the `protoc` binary should have been installed as well. If it hasn't been installed, you can run the following commands to install it.

File: src/php/README.md [hunk 6]
```diff
@@ -79,58 +88,96 @@ $ make
...
 
 ```sh
-$ cd grpc/src/php
-$ ./bin/run_tests.sh
+$ cd grpc/third_party/protobuf
+$ sudo make install   # 'make' should have been run by core grpc
 ```
 
-## Generated Code Tests
+Or you can download a `protoc` binaries from [here](https://github.com/google/protobuf/releases).
```
Comment (LisaFC) [inline src/php/README.md]: Alternatively,  you can download `protoc` binaries from [the protocol buffers Github repository](https://github.com/google/protobuf/releases).

File: src/php/README.md [hunk 7]
```diff
@@ -79,58 +88,96 @@ $ make
...
+
 
-Install `protoc-gen-php`
+### PHP protobuf compiler
+
+You need to install `protoc-gen-php`, so that you can generate stub classes `.php` files from service definition `.proto` files.
```
Comment (LisaFC) [inline src/php/README.md]: You need to install `protoc-gen-php` to generate stub class `.php` files from service definition `.proto` files.

File: src/php/README.md [hunk 8]
```diff
@@ -79,58 +88,96 @@ $ make
...
 ```sh
 $ cd grpc/src/php
 $ ./bin/generate_proto_php.sh
 ```
 
-Run a local server serving the math services
+### Run test server
 
- - Please see [Node][] on how to run an example server
+Run a local server serving the math services. Please see [Node][] on how to run an example server
```
Comment (LisaFC) [inline src/php/README.md]: on how to run an example server -> for how to run an example server.

**Comment by stanley-cheung** [comment]
Thanks @LisaFC for all the reviews! I have fixed all your suggestions. PTAL again, thanks!

**Comment by LisaFC** [comment]
Looks great!

**Pull request merged**